### PR TITLE
Raw arrays - SkillDatabase

### PR DESCRIPTION
**Pull request by fredg1**

### CODE REVIEW

**Review by gausie — APPROVED** [review]

File: src/net/sourceforge/kolmafia/persistence/SkillDatabase.java [hunk 1]
```diff
@@ -1532,28 +1532,28 @@ public static final boolean contains(final String skillName) {
   }
 
   public static final void generateSkillList(final StringBuffer buffer, final boolean appendHTML) {
-    ArrayList<String>[] categories = new ArrayList[SkillDatabase.CATEGORIES.length];
+    List<ArrayList<String>> categories = new ArrayList<>(SkillDatabase.CATEGORIES.length);
```
Comment (gausie) [inline src/net/sourceforge/kolmafia/persistence/SkillDatabase.java:1535]: Can this just be `List<List<` or is the fact that its `ArrayList` important?

Comment (fredg1) [reply src/net/sourceforge/kolmafia/persistence/SkillDatabase.java:1535]: SkillDatabase.appendSkillList(StringBuffer, boolean, String, ArrayList<String>) is called at the end

Comment (fredg1) [reply src/net/sourceforge/kolmafia/persistence/SkillDatabase.java:1535]: (so yes, it's important to note that they are ArrayLists)

Comment (gausie) [reply src/net/sourceforge/kolmafia/persistence/SkillDatabase.java:1535]: But does *that* need to specifically be an ArrayList?

Comment (fredg1) [reply src/net/sourceforge/kolmafia/persistence/SkillDatabase.java:1535]: Uhhhhh... no!

**Comment by fredg1** [comment]
(wrong button, didn't mean to start a review...)

File: src/net/sourceforge/kolmafia/persistence/SkillDatabase.java (hunk 1 above)
Comment (fredg1) [reply src/net/sourceforge/kolmafia/persistence/SkillDatabase.java]: Uhhhhh... no!

**Comment by gausie** [comment]
Nice. I think people have sometimes typed things too closely - you basically never need to type something as `ArrayList` or `LinkedList` specifically (nor `HashMap`, `SortedMap` etc)

**Review by gausie — APPROVED** [review]

**Pull request merged**